### PR TITLE
Remove login URL from external wallet structure

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/33884

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by Douglashdaniel — APPROVED** [review]
++ Wallet Front-end

**Review by emerick — APPROVED** [review]

File: android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java [hunk 1]
```diff
@@ -411,10 +411,9 @@ private void web3ButtonClick(View view) {
     private void sendTipButtonClick(View view) {
         mSendButton.setOnClickListener(v -> {
             if (mIsLogoutState) {
-                if (mExternalWallet != null) {
-                    TabUtils.openUrlInNewTab(false, mExternalWallet.getLoginUrl());
-                    dismissRewardsPanel();
```
Comment (szilardszaloki) [inline android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java:416]: Is this no longer needed?

Comment (zenparsing) [reply android/java/org/chromium/chrome/browser/rewards/tipping/RewardsTippingPanelFragment.java:416]: Good catch - thanks!

File: components/brave_rewards/core/rewards_callbacks.h [hunk 2]
```diff
@@ -23,6 +23,9 @@ using AttestPromotionCallback = mojom::RewardsEngine::AttestPromotionCallback;
 
 using ClaimPromotionCallback = mojom::RewardsEngine::ClaimPromotionCallback;
 
+using BeginExternalWalletLoginCallback =
+    mojom::RewardsEngine::BeginExternalWalletLoginCallback;
+
```
Comment (szilardszaloki) [inline components/brave_rewards/core/rewards_callbacks.h:28]: Nit: can we move this up under `AttestPromotionCallback` (to keep the alphabetical order)?

File: components/brave_rewards/core/wallet_provider/connect_external_wallet.h [hunk 3]
```diff
@@ -58,6 +66,7 @@ class ConnectExternalWallet {
   void CheckLinkageCallback(endpoints::GetWallet::Result&& result);
 
   base::RetainingOneShotTimer linkage_checker_;
+  OAuthInfo oauth_info_;
```
Comment (szilardszaloki) [inline components/brave_rewards/core/wallet_provider/connect_external_wallet.h:69]: I think this should be `protected`, and then `Authorize()` wouldn't need the `const OAuthInfo& oauth_info` parameter.

Comment (szilardszaloki) [reply components/brave_rewards/core/wallet_provider/connect_external_wallet.h:69]: Nor `GetOAuthLoginURL()`.

File: components/brave_rewards/common/mojom/rewards.mojom [hunk 4]
```diff
@@ -412,16 +412,17 @@ struct ExternalWallet {
...
   string member_id;
   string account_url;
-  string login_url;
   string activity_url;
   map<string, double> fees;
 };
 
+struct ExternalWalletLoginParams {
+  string url;
+};
```
Comment (szilardszaloki) [inline components/brave_rewards/common/mojom/rewards.mojom:424]: Do we need a separate Mojom struct for this URL?

Comment (zenparsing) [reply components/brave_rewards/common/mojom/rewards.mojom:424]: The intension here is to leave room for `BeginLogin` to return headers (or cookies) to support other login flows.

**Review by deeppandya — APPROVED** [review]
Android changes LGTM

**Review by szilardszaloki — APPROVED** [review]

**Pull request merged**